### ISSUE TITLE
useSessionStorage causing React+Nextjs application crash. Error "Hydration failed because the initial UI does not match what was rendered on the server"

### ISSUE THREAD

**Issue by roker15** [juliencrn/usehooks-ts#176]
My application was working fine. I was required to refresh pages without losing state. so i replaced `useState `with `useSessionStorage `wherever required. Now it throwing error
`Hydration failed because the initial UI does not match what was rendered on the server`

This [Link](https://nextjs.org/docs/messages/react-hydration-error) might help but not sure.

**Comment by harrywebdev** [comment]
I reckon this is most likely because the `useState` hook doesn't use the provided `initialValue` here
https://github.com/juliencrn/usehooks-ts/blob/master/src/useSessionStorage/useSessionStorage.ts#L39

which is, what is expected on client on the first render. I think the line should read

```
const [storedValue, setStoredValue] = useState<T>(initialValue)
```

and if I understand the code correctly, then this comment

> // Pass initial state function to useState so logic is only executed once

is actually handled by the `useEffect(..., [])` here
https://github.com/juliencrn/usehooks-ts/blob/master/src/useSessionStorage/useSessionStorage.ts#L69

I'll see if I can create a PR - maybe in case @juliencrn or @createdbymahmood think that's a sound idea/fix?

**Comment by juliencrn** [comment]
Should have been fixed by #435, will be in the next release. Thanks all!
